zone/client_packet.cpp

```diff
@@ -16533,3 +16644,306 @@ void Client::ReloadExpansionProfileSetting()
+		gt.guild_id = GuildID();
+		gt.tribute_id_1 = data->tribute_id_1;
+		gt.tribute_id_2 = data->tribute_id_2;
+		gt.tribute_id_1_tier = data->tribute_1_tier;
+		gt.tribute_id_2_tier = data->tribute_2_tier;
+		gt.enabled = 0;
+		gt.time_remaining = RuleI(Guild, TributeTime);
+		GuildTributesRepository::ReplaceOne(database, gt);
+
+		guild->tribute.enabled = 0;
+		guild->tribute.id_1 = data->tribute_id_1;
+		guild->tribute.id_2 = data->tribute_id_2;
+		guild->tribute.id_1_tier = data->tribute_1_tier;
+		guild->tribute.id_2_tier = data->tribute_2_tier;
+		guild->tribute.time_remaining = RuleI(Guild, TributeTime);
+
+		ServerPacket* sp = new ServerPacket(ServerOP_GuildTributeUpdate, sizeof(GuildTributeUpdate));
+		GuildTributeUpdate* gtu = (GuildTributeUpdate*)sp->pBuffer;
+
+		gtu->guild_id = GuildID();
+		gtu->tribute_id_1 = data->tribute_id_1;
+		gtu->tribute_id_2 = data->tribute_id_2;
+		gtu->tribute_id_1_tier = data->tribute_1_tier;
+		gtu->tribute_id_2_tier = data->tribute_2_tier;
+		gtu->enabled = 0;
+		gtu->favor = guild->tribute.favor;
+		gtu->time_remaining = RuleI(Guild, TributeTime);
+
+		worldserver.SendPacket(sp);
+		safe_delete(sp);
+	}
+}
+
+void Client::Handle_OP_GuildTributeToggle(const EQApplicationPacket* app)
+{
+	LogTribute("Received OP_Handle_OP_GuildTributeToggle of length [{}]", app->size);
+
+	if (app->size != sizeof(GuildTributeToggleReq_Struct)) {
+		LogError("Invalid size on Handle_OP_GuildModifyBenefits packet");
+	}
+	else {
+		GuildTributeToggleReq_Struct* gt = (GuildTributeToggleReq_Struct*)app->pBuffer;
+
+		ServerPacket* sp = new ServerPacket(ServerOP_GuildTributeActivate, sizeof(GuildTributeUpdate));
+		GuildTributeUpdate* gtu = (GuildTributeUpdate*)sp->pBuffer;
+		auto guild = guild_mgr.GetGuildByGuildID(GuildID());
+		if (guild) {
+			switch (gt->command) {
+			case 0: {
+				gtu->guild_id = GuildID();
+				gtu->enabled = 0;
+				gtu->favor = guild->tribute.favor;
+				gtu->tribute_id_1 = guild->tribute.id_1;
+				gtu->tribute_id_2 = guild->tribute.id_2;
+				gtu->tribute_id_1_tier = guild->tribute.id_1_tier;
+				gtu->tribute_id_2_tier = guild->tribute.id_2_tier;
+				gtu->time_remaining = guild->tribute.time_remaining;
+
+				worldserver.SendPacket(sp);
+				break;
+			}
+			case 1: {
+				gtu->guild_id = GuildID();
+				gtu->enabled = 1;
+				gtu->favor = guild->tribute.favor;
+				gtu->tribute_id_1 = guild->tribute.id_1;
+				gtu->tribute_id_2 = guild->tribute.id_2;
+				gtu->tribute_id_1_tier = guild->tribute.id_1_tier;
+				gtu->tribute_id_2_tier = guild->tribute.id_2_tier;
+				gtu->time_remaining = guild->tribute.time_remaining;
+
+				worldserver.SendPacket(sp);
+				break;
+			}
+			default: {
+				LogError("Unknown GuildTributeUpdate to WorldServer command received.");
+				break;
+			}
+			}
+		}
+		safe_delete(sp);
+	}
+	return;
+}
+
+void Client::Handle_OP_GuildTributeDonateItem(const EQApplicationPacket* app)
+{
+	LogTribute("Received OP_Handle_OP_GuildTributeDonateItem of length [{}]", app->size);
```

Comment:
We get this logging for free with our existing packet logging, can remove

Reply:
Removed.